forest/bonsai/simulate_gps_data.py

```diff
@@ -170,27 +181,77 @@ def bounding_box(center: Tuple[float, float], radius: int) -> Tuple:
     return lat - lat_const, lon - lon_const, lat + lat_const, lon + lon_const
 
 
-@dataclass
 class Attributes:
-    """This class holds the attributes needed to create an instance of a person
+    """This class holds the attributes needed to create an instance of a
+    Person class"""
 
-    Args:
-        vehicle used for distances and time of flights
-        main_occupation used for routine action in weekdays
-        active_status = 0-10
-            used for probability in free time to take an action
-            or stay home
-        travelling status = 0-10
-            used to derive amount of distance travelled
-        preferred_places  = [x1, x2, x3]
-            used to sample action when free time
-            where x1-x3 are amenities (str)
-    """
-    vehicle: Vehicle
-    main_occupation: Occupation
-    active_status: int
-    travelling_status: int
-    preferred_places: list
+    def __init__(self, vehicle: Optional[str] = None,
+                 main_employment: Optional[str] = None,
+                 active_status: Optional[int] = None,
+                 travelling_status: Optional[int] = None,
+                 preferred_places: Optional[List[str]] = None, **kwargs):
+        """Error check and generate missing data for attributes
+
+        Args:
+            vehicle: used for distances and time of flights
+            main_occupation: used for routine action in weekdays
+            active_status: used for probability in free time to take an action
+                or stay home
+            travelling status: used to derive amount of distance travelled
+            preferred_places :used to sample action when free time
+                where x1-x3 are amenities (str)
+        Raises:
+            ValueError: for incorrect vehicle type
+            ValueError: for incorrect main_employment type
+            ValueError: if active_status is not between 0 and 10
+            ValueError: if travelling_status is not between 0 and 10
+            ValueError: if an exit from possible_exits is not correct type
+        """
+        if vehicle is not None:
+            self.vehicle = Vehicle[vehicle.upper()]
```

Comment:
This would actually raise `KeyError` in case of a wrong vehicle type. Why not simply: `Vehicle(vehicle)`? This is also more flexible in case vehicle strings ever change.